### PR TITLE
[WIP] daemon improvements

### PR DESCRIPTION
**Pull request by ngortheone**
Code readability and consistency improvements. The change is broken down into small commits, each commit has a commit message with rationale for the change.

cc @kevans91 @bsdimp

### CODE REVIEW

**Review by igalic — CHANGES_REQUESTED** [review]
this (a46e046) is, unfortunately, running counter to our style(9) definition

File: usr.sbin/daemon/daemon.c [hunk 1]
```diff
@@ -670,24 +682,21 @@ handle_chld(int signo __unused)
...
 }
 
 static void
 reopen_log(struct log_params *lpp)
 {
-	int outfd;
-
 	do_log_reopen = 0;
-	outfd = open_log(lpp->outfn);
+	int outfd = open_log(lpp->outfn);
```
Comment (kevans91) [inline usr.sbin/daemon/daemon.c:699]: This one's a no-go, declarations must be at the top of the containing block.

File: usr.sbin/daemon/daemon.c [hunk 2]
```diff
@@ -139,27 +141,43 @@ usage(int exitcode)
...
+        int dosyslog = 0;
+        int child_eof = 0;
+        int stdmask = STDOUT_FILENO | STDERR_FILENO;
+        int logpri = LOG_NOTICE;
+        int logfac = LOG_DAEMON;
+        int log_reopen = 0;
+        int pfd[2] = { -1, -1 };
+        int outfd = -1;
+
+        sigset_t mask_susp;
```
Comment (kevans91) [inline usr.sbin/daemon/daemon.c:165]: Same comment as below about declarations

File: usr.sbin/daemon/daemon.c [hunk 3]
```diff
@@ -139,27 +141,43 @@ usage(int exitcode)
...
+	sigset_t mask_orig;
+
+	sigset_t mask_read;
+	sigemptyset(&mask_read);
+
+	sigset_t mask_term;
+	sigemptyset(&mask_term);
+
+
+        struct log_params logpar = { 0 };
```
Comment (kevans91) [inline usr.sbin/daemon/daemon.c:186]: Same comment as below about declarations

File: usr.sbin/daemon/daemon.c [hunk 4]
```diff
@@ -292,22 +301,15 @@ main(int argc, char *argv[])
 	 * and syslog.
 	 */
 	if (pidfile || ppidfile || restart || outfd != -1 || dosyslog) {
-		struct sigaction act_term, act_chld, act_hup;
```
Comment (kevans91) [inline usr.sbin/daemon/daemon.c:271]: I have mixed feelings about this one, given that they're really only used within this block

**Comment by kevans91** [comment]
I don't think I understand the last one; daemon shouldn't be running forever in the non-forking mode, it just doesn't need to monitor the spawned process so it jumps right into `execvp`  (note that `pid` will not be reassigned from -1 if none of these options are set).

**Comment by ngortheone** [comment]
@kevans91 disregard my last commit, I misinterpreted the code a bit. I will revert it soon.
But it does leave a big question open, which I will share in email

**Pull request closed without merging**